xarray/core/alignment.py

```diff
@@ -324,7 +325,8 @@ def assert_no_index_conflict(self) -> None:
                     "- they may be used to reindex data along common dimensions"
                 )
 
-    def _need_reindex(self, dims, cmp_indexes) -> bool:
+    @deprecate_dims
+    def _need_reindex(self, dim, cmp_indexes) -> bool:
```

Comment:
do we need to deprecate here at all? This is purely internal...

Reply:
Yeah good point, let me remove these